virtual-server/examples/go/main.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		log.Fatalf("Failed to create client\n")
+	}
+
+	vsv1alpha.AddToScheme(c.Scheme())
+
+	// prepare config for kubevirt client, you need to set env variable, KUBECONFIG=<path-to-kubeconfig>/.kubeconfig
+	kubevirtClientConfig := kubecli.DefaultClientConfig(&pflag.FlagSet{})
+
+	// get the kubevirt client, using which kubevirt resources can be managed.
+	kubevirtClient, err := kubecli.GetKubevirtClientFromClientConfig(kubevirtClientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain KubeVirt client: %v\n", err)
+	}
+
+	// Create a new VirtualServer with the name "my-virtual-server" to be deployed in the "default" namespace
+	virtualServer := vsv1alpha.NewVirtualServer(name, namespace)
+	// Set the region the VirtualServer will be deployed to
+	virtualServer.SetRegion("ORD1")
+	// Specify the VirtualServer operating system
+	virtualServer.SetOS(vsv1alpha.VirtualServerOSTypeLinux)
+	// Set a GPU type request for the VirtualServer
+	virtualServer.SetGPUType("Quadro_RTX_4000")
+	// Set the number of GPUs to request for the VirtualServer
+	virtualServer.SetGPUCount(1)
+	// Set the cpu core count for the VirtualServer
+	virtualServer.SetCPUCount(2)
+	//virtualServer.SetCPUType("amd-epyc-rome")
+	// Set the memory request for the VirtualServer
+	virtualServer.SetMemory("2Gi")
+	// Add a user to be added by cloudinit
+	virtualServer.AddUser(vsv1alpha.VirtualServerUser{
+		Username: "myuser",
+		Password: "password",
+	})
+	// Configure the root filesystem of the VirtualServer to clone a preexisting PVC namedubuntu1804-docker-master-20210210-ord1
+	// sourced in the vd-images namespace
+	err = virtualServer.ConfigureStorageRootWithPVCSource(vsv1alpha.VirtualServerStorageRootPVCSource{
+		Size:             "40Gi",
+		PVCName:          "ubuntu1804-docker-master-20210210-ord1",
+		PVCNamespace:     "vd-images",
+		StorageClassName: "ceph-ssd-2-replica",
+		VolumeMode:       corev1.PersistentVolumeBlock,
+		AccessMode:       corev1.ReadWriteOnce,
+	})
+	if err != nil {
+		log.Fatalf("Cound not configure root filesystem\n")
+	}
+	// Expose tcp ports 22 and 443 on the VirtualServer
+	virtualServer.ExposeTCPPorts([]int32{22, 443})
+	// Expose a single udp port 4172 on the VirtualServer
+	virtualServer.ExposeUDPPort(4172)
+	// Set the VirtualServer to start as soon as it is created
+	virtualServer.InitializeRunning(true)
+
+	// Delete virtual server if already exists
+	err = c.Delete(context.Background(), virtualServer)
+	if err != nil {
+		if apierrors.IsNotFound(err) == true {
+			fmt.Printf("VirtualServer %s in namespace %s already deleted", name, namespace)
+		} else {
+			log.Fatalf("Failed to create VirtualServer\nReason: %s", err.Error())
+		}
+	}
+
+	// Create new virtual server
+	err = c.Create(context.Background(), virtualServer)
+	if err != nil {
+		log.Fatalf("Failed to create VirtualServer\nReason: %s", err.Error())
+	}
+
+	// Wait until virtual server is ready
+	fmt.Printf("VirtualServer status: %s", Ready(namespace, name, c))
```

Comment:
I'm not sure this will work as expected. If the VS has yet to be created yet (for whatever reason) or has not been reconciled (so that the conditions are not yet present) this will return "return" or throw a fatal log and kill the process. When I think of Ready, I think of a function that blocks until it's ready, or takes a callback.

Reply:
Optionally, it could also block/callback for a specific VirtualServerConditionType with a status or a VirtualServerConditionReason.

Reply:
It works. The `Ready()` function is blocked until the required condition is fulfilled.